core/src/main/java/io/fabric8/maven/core/service/openshift/ImageStreamService.java

```diff
@@ -230,4 +238,19 @@ private String findTagSha(OpenShiftClient client, String imageStreamName, String
             throw new MojoExecutionException("Could not find a tag in the ImageStream " + imageStreamName);
         }
     }
+
+    private boolean isDate1AfterDate2(String date1, String date2) {
+        SimpleDateFormat dateFormatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+
+        if(Strings.isNotBlank(date1) || Strings.isNotBlank(date2)) {
```

Comment:
@sthaha @hrishin This statement looks like not correct to me. Maybe I am wrong. Can you please check.

Reply:
Good catch!
Agree, I think we will hit this condition all the time, don't we?

Reply:
Yaa, I also think the same. Only when both are blank then we are going further. It seems like this to me.

Reply:
This check seems redundant to me. Even if `SimpleDateFormat` fails to parse the string; it should throw ParseException.

Reply:
Format needs to be `yyyy-MM-dd'T'HH:mm:ss'Z'`